lib/daryl_inject.rb

```diff
@@ -0,0 +1,19 @@
+class Array
+
+  def daryl_inject(default = self.first, &block)
+
+    return default if self.length == 1
+    self.shift if default == self.first
+    self.each { |x| default = block.call(default, x) }
+    default
+
+  end
+
+  def daryl_inject2(result = self.first, &block)
```

Comment:
Unused method argument - `result`. If it's necessary, use `_` or `_result` as an argument name to indicate that it won't be used.